### PR TITLE
[.github] - refacto: migrate to container registry

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR aims at moving all deployments to build and push images to the Artifact Registry.

It also updates the `revert-x` workflows. The process is now the following:
- Run the [List Image Tags](https://github.com/dust-tt/dust/actions/workflows/list-tags.yaml) workflow for the desired service
- Inspect the list of available image tags (quite fast to run) and copy the one you want. Note: image tags are commit sha.
- Trigger the `revert-x` workflow with the desired image tag.

Unfortunately GH doesn't support human input between to steps so we have to run the tag listing step as a separate workflow. We can probably improve this process later, by interfacing through Slack. But for now it gives to flexibility to rollback to a selected hash (which has proven to be convenient).

## Risk

Break CI but deployments can still be done manually if any emergency

### CODE REVIEW

**Comment by JulesBelveze** [comment]
Note: deployment workflow was tested for front-edge

**Review by spolu — COMMENTED** [review]
Looks reasonable but left a few questions

File: .github/workflows/revert-core.yml [hunk 1]
```diff
@@ -1,7 +1,12 @@
-name: Revert Core
+name: Deploy Core
 
 on:
   workflow_dispatch:
+    inputs:
+      image_tag:
```
[the remark this answers is not in the document]
Comment (JulesBelveze) [reply .github/workflows/revert-core.yml:6]: Yeah, messed up with the files sorry, should be good now 👌🏼

File: k8s/cloudbuild.yaml [hunk 2]
```diff
@@ -1,25 +1,19 @@
...
-      - --build-arg
-      - NEXT_PUBLIC_DUST_CLIENT_FACING_URL=${_DUST_CLIENT_FACING_URL}
-      - .
+  - id: 'Build Container Image'
+    name: 'ghcr.io/depot/cli:latest'
+    script: |
+      depot build \
+        --project 3vz0lnf16v \
+        --provenance=false \
+        -t us-central1-docker.pkg.dev/${PROJECT_ID}/dust-images/${_IMAGE_NAME}:${SHORT_SHA} \
```
[the remark this answers is not in the document]
Comment (JulesBelveze) [reply k8s/cloudbuild.yaml:8]: Yeah I'll leave it as is now and will add an dropdown input to select the region to deploy to.

[the remark this answers is not in the document]
Comment (JulesBelveze) [reply k8s/cloudbuild.yaml:8]: Will tackle this once this one is merged

**Review by spolu — APPROVED** [review]
LGTM

**Pull request merged**